flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/services/BigQueryServicesImpl.java

```diff
@@ -165,6 +176,54 @@ public void close() {
         }
     }
 
+    /** Implementation of a BigQuery write client wrapper. */
+    public static class StorageWriteClientImpl implements StorageWriteClient {
+        private static final HeaderProvider USER_AGENT_HEADER_PROVIDER =
+                FixedHeaderProvider.create(
+                        "user-agent", "Apache_Flink_Java/" + FlinkVersion.current().toString());
+
+        private final BigQueryWriteClient client;
+
+        private StorageWriteClientImpl(CredentialsOptions options) throws IOException {
+            BigQueryWriteSettings.Builder settingsBuilder =
+                    BigQueryWriteSettings.newBuilder()
+                            .setCredentialsProvider(
+                                    FixedCredentialsProvider.create(options.getCredentials()))
+                            .setTransportChannelProvider(
+                                    BigQueryReadSettings.defaultGrpcTransportProviderBuilder()
+                                            .setHeaderProvider(USER_AGENT_HEADER_PROVIDER)
+                                            .build());
+
+            this.client = BigQueryWriteClient.create(settingsBuilder.build());
+        }
+
+        @Override
+        public StreamWriter createStreamWriter(String streamName, ProtoSchema protoSchema)
+                throws IOException {
+            RetrySettings retrySettings =
+                    RetrySettings.newBuilder()
+                            .setMaxAttempts(5)
+                            .setTotalTimeout(Duration.ofMinutes(5))
+                            .setInitialRpcTimeout(Duration.ofSeconds(30))
+                            .setMaxRpcTimeout(Duration.ofMinutes(2))
+                            .setRpcTimeoutMultiplier(1.6)
+                            .setRetryDelayMultiplier(1.6)
+                            .setInitialRetryDelay(Duration.ofMillis(1250))
+                            .setMaxRetryDelay(Duration.ofSeconds(5))
```

Comment:
can we document all of these hardcoded values?

Reply:
Yes, we plan to update the root README with all relevant details once implementation specific PRs are merged. Buganizer task `b/330779130` exists for internally tracking this.

Reply:
i think it'd be helpful to have in-line comments here. if i was reading this code, i wouldnt think to look at a readme to figure out what these values mean

Reply:
thanks! you can resolve this conversation